src/Umbraco.Web.BackOffice/Trees/TreeController.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Concurrent;
+using Microsoft.Extensions.DependencyInjection;
+using Umbraco.Core;
+using Umbraco.Core.Services;
+using Umbraco.Web.Trees;
+
+namespace Umbraco.Web.BackOffice.Trees
+{
+    /// <summary>
+    /// The base controller for all tree requests
+    /// </summary>
+    public abstract class TreeController : TreeControllerBase
+    {
+        private static readonly ConcurrentDictionary<Type, TreeAttribute> _treeAttributeCache = new ConcurrentDictionary<Type, TreeAttribute>();
+
+        private readonly TreeAttribute _treeAttribute;
+
+        private readonly ILocalizedTextService _textService;
+
+        protected TreeController()
+        {
+            var serviceProvider = HttpContext.RequestServices;
```

Comment:
Wasn't too sure about this - should we support a parameterless constructor to make things easier for package/solution developers (and perhaps other core trees)?  Or is it better to expect them to use a constructor where the dependencies (i.e. `ILocalizedTextService`) are provided?

Reply:
For now, we remove the parameterless constructors. These make testing harder, and package developers should be interested in having testable code. That said, maybe we reintroduce them later. For now, it is most important, that we do not use these untestable parts ourselves.